server.go

```diff
@@ -186,6 +235,248 @@ func main() {
 	}
 }
 
+const msgUpstreamParticipantCreateError = "could not create upstream participant. response status: %s"
+
+func upstreamNewParticipant(c echo.Context, p participant) (id string, err error) {
+	item := leaderboard_item{}
+	item.UserName = p.GitHubName
+	item.Slug = p.GitHubName
+	item.Score = 0
+	item.Archived = false
+	item.Draft = false
+
+	payload := leaderboard_payload{}
+	payload.Fields = item
+
+	body, err := json.Marshal(payload)
+	if err != nil {
+		return
+	}
+
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/collections/%s/items?live=true", webflowBaseAPI, webflowCollection), bytes.NewReader(body))
+	if err != nil {
+		return
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", webflowToken))
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("accept-version", "1.0.0")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return
+	}
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		c.Logger().Debug(req)
+		var responseBody []byte
+		_, _ = res.Body.Read(responseBody)
+		c.Logger().Debug(responseBody)
+		// return a real error to the caller to indicate failure
+		errMsg := fmt.Sprintf(msgUpstreamParticipantCreateError, res.Status)
+		if errContext := c.String(http.StatusInternalServerError, errMsg); errContext != nil {
+			err = errContext
+		} else {
+			err = fmt.Errorf(errMsg)
+		}
+		return
+	}
+
+	var response leaderboard_response
+	err = json.NewDecoder(res.Body).Decode(&response)
+	if err != nil {
+		return
+	}
+	id = response.Id
+	return
+}
+
+const msgUpstreamParticipantUpdateError = "could not update score. response status: %s"
```

Comment:
Same comment as on the other error constant above, buyers choice though!

Reply:
tried 'error types' here: a751930
please holler if it could be done better